src/index.ts

```diff
@@ -35,8 +35,9 @@ export interface Options {
  * are Buffers with the contents of the certificate private key and certificate
  * file, respectively
  */
-export async function certificateFor(domain: string, options: Options = {}) {
-  debug(`Certificate requested for ${ domain }. Skipping certutil install: ${ Boolean(options.skipCertutilInstall) }. Skipping hosts file: ${ Boolean(options.skipHostsFile) }`);
+export async function certificateFor(domains: string | string[], options: Options = {}) {
+  const domain = Array.isArray(domains) ? domains[0] : domains;
```

Comment:
I appreciate the backwards-compatibility efforts here. I recommend that you simplify the following code by making `domains` **always an array** and then using `domains` only for the rest of the code. Then you can remove the conditionals for when it's one domain.
```diff
-export async function certificateFor(domains: string | string[], options: Options = {}) {
-  const domain = Array.isArray(domains) ? domains[0] : domains;
+export async function certificateFor(requestedDomains: string | string[], options: Options = {}) {
+  const domains = Array.isArray(requestedDomains) ? requestedDomains : [requestedDomains]
```

If you do that, you can rename all the other private identifiers to plurals so the code is easier to read, so `pathForDomain` becomes `pathForDomains` et cetera. This makes the types easier to follow!!

Reply:
Cool @zetlen , I thought that too I just wanted to save the backward compatibility. If we're not bothered about backward compatibility (it's really not that big of a deal if we do a major version bump) I'll chang ethis